features/liquidity/components/BondLiquidityDialog.tsx

```diff
@@ -26,23 +24,37 @@ import React, { useEffect, useRef, useState } from 'react'
 import { toast } from 'react-hot-toast'
 import { formatSdkErrorMessage } from 'util/formatSdkErrorMessage'
 
-import { usePoolTokensDollarValue, useStakedTokenBalance } from '../hooks'
+import { useQueryPoolLiquidity } from '../../../queries/useQueryPools'
+import { usePoolTokensDollarValue } from '../hooks'
 import { LiquidityInputSelector } from './LiquidityInputSelector'
 import { PercentageSelection } from './PercentageSelection'
 import { StakingSummary } from './StakingSummary'
 import { StateSwitchButtons } from './StateSwitchButtons'
 
-export const BondLiquidityDialog = ({ isShowing, onRequestClose, poolId }) => {
+type BondLiquidityDialogProps = {
+  isShowing: boolean
+  onRequestClose: () => void
+  poolId: string
+}
+
+export const BondLiquidityDialog = ({
+  isShowing,
+  onRequestClose,
+  poolId,
+}: BondLiquidityDialogProps) => {
   const [dialogState, setDialogState] = useState<'stake' | 'unstake'>('stake')
 
-  const tokenA = useBaseTokenInfo()
-  const tokenB = useTokenInfoByPoolId(poolId)
+  const [pool] = useQueryPoolLiquidity({
+    poolId,
+  })
 
-  const [{ myLiquidity } = {} as any] = usePoolLiquidity({ poolId })
-  const [stakedAmount] = useStakedTokenBalance({ poolId })
+  const { pool_assets, liquidity } = pool || {}
+  const [tokenA, tokenB] = pool_assets || []
 
   const maxLiquidityTokenAmount =
-    dialogState === 'stake' ? myLiquidity?.tokenAmount ?? 0 : stakedAmount ?? 0
+    dialogState === 'stake'
+      ? liquidity?.fluid.provided.tokenAmount ?? 0
```

Comment:
can we change the word fluid? Maybe available is a better term? open to suggestions